tests/unit/lifecycle/LifecycleTask.spec.js

```diff
@@ -0,0 +1,340 @@
+    * @return {undefined}
+    */
+    addAbortMPU(days) {
+        this.abortMPU = { DaysAfterInitiation: days };
+        return this;
+    }
+}
+
+// get all rule ID's
+function getRuleIDs(rules) {
+    return rules.map(rule => rule.ID).sort();
+}
+
+describe('lifecycle task helper methods', () => {
+    let lct;
+
+    before(() => {
+        const lp = {
+            getStateVars: () => (
+                {
+                    enabledRules: {
+                        expiration: { enabled: true },
+                        transitions: { enabled: true },
+                        noncurrentVersionTransitions: { enabled: true },
+                        noncurrentVersionExpiration: { enabled: true },
+                        abortIncompleteMultipartUpload: { enabled: true },
+                    },
+                }
+            ),
+        };
+        lct = new LifecycleTask(lp);
+    });
+
+    describe('_filterRules for listObjectsV2 contents',
+    () => {
+        it('should filter out Status disabled rules', () => {
+            const mBucketRules = [
+                new Rule().addID('task-1').build(),
+                new Rule().addID('task-2').disable().build(),
+                new Rule().addID('task-3').build(),
+                new Rule().addID('task-4').build(),
+                new Rule().addID('task-2').disable().build(),
+            ];
+            const item = {
+                Key: 'example-item',
+                LastModified: pastDate,
+            };
+            const objTags = { TagSet: [] };
+
+            const res = lct._filterRules(mBucketRules, item, objTags);
+            const expected = mBucketRules.filter(rule =>
+                rule.Status === 'Enabled');
+            assert.deepStrictEqual(getRuleIDs(res), getRuleIDs(expected));
+        });
+
+        it('should filter out unmatched prefixes', () => {
+            const mBucketRules = [
+                new Rule().addID('task-1').addPrefix('atask').build(),
+                new Rule().addID('task-2').addPrefix('atasker').build(),
+                new Rule().addID('task-3').addPrefix('cat-').build(),
+                new Rule().addID('task-4').addPrefix('xatask').build(),
+                new Rule().addID('task-5').addPrefix('atasr').build(),
+                new Rule().addID('task-6').addPrefix('Atask').build(),
+                new Rule().addID('task-7').addPrefix('atAsk').build(),
+                new Rule().addID('task-8').build(),
+            ];
+            const item1 = {
+                Key: 'atasker-example-item',
+                LastModified: currentDate,
+            };
+            const item2 = {
+                Key: 'cat-test',
+                LastModified: currentDate,
+            };
+            const objTags = { TagSet: [] };
+
+            const res1 = lct._filterRules(mBucketRules, item1, objTags);
+            assert.strictEqual(res1.length, 3);
+            const expRes1 = getRuleIDs(mBucketRules.filter(rule =>
```

Comment:
In this test, it looks like you're just testing the test functions, is it what you want? As `res1` is not checked (same for `res2` below).

Reply:
oops my mistake, ty meant to test against `res1`